module/mempool/herocache/backdata/heropool/pool_test.go

```diff
@@ -287,20 +287,19 @@ func testInvalidatingHead(t *testing.T, pool *Pool, entities []*unittest.MockEnt
 			// used tail should point to the last element in pool, since we are
 			// invalidating head.
 			require.Equal(t, entities[totalEntitiesStored-1].ID(), usedTail.id)
-			require.Equal(t, EIndex(totalEntitiesStored-1), pool.used.tail.getSliceIndex())
+			require.Equal(t, EIndex(totalEntitiesStored-1), pool.used.tail)
 
 			// used head must point to the next element in the pool,
 			// i.e., invalidating head moves it forward.
 			require.Equal(t, entities[i+1].ID(), usedHead.id)
-			require.Equal(t, EIndex(i+1), pool.used.head.getSliceIndex())
+			require.Equal(t, EIndex(i+1), pool.used.head)
 		} else {
 			// pool is empty
 			// used head and tail must be nil and their corresponding
 			// pointer indices must be undefined.
 			require.Nil(t, usedHead)
 			require.Nil(t, usedTail)
-			require.True(t, pool.used.tail.isUndefined())
-			require.True(t, pool.used.head.isUndefined())
```

Comment:
Shoudn't these two conditions be like the following? 
```go
			require.Zero(t, pool.used.tail)
			require.Zero(t, pool.used.head)
```

Reply:
This question is addressed in comments below. I will take an action depending on our discussion in the comments below.